lntest/itest/lnd_multi-hop_test.go

```diff
@@ -366,3 +369,1607 @@ func createThreeHopNetwork(ht *lntemp.HarnessTest,
+	payHash := preimage.Hash()
+	invoiceReq := &invoicesrpc.AddHoldInvoiceRequest{
+		Value:      invoiceAmt,
+		CltvExpiry: finalCltvDelta,
+		Hash:       payHash[:],
+	}
+	carolInvoice := carol.RPC.AddHoldInvoice(invoiceReq)
+
+	// Subscribe the invoice.
+	stream := carol.RPC.SubscribeSingleInvoice(payHash[:])
+
+	// Now that we've created the invoice, we'll send a single payment from
+	// Alice to Carol. We won't wait for the response however, as Carol
+	// will not immediately settle the payment.
+	req := &routerrpc.SendPaymentRequest{
+		PaymentRequest: carolInvoice.PaymentRequest,
+		TimeoutSeconds: 60,
+		FeeLimitMsat:   noFeeLimitMsat,
+	}
+	alice.RPC.SendPayment(req)
+
+	// At this point, all 3 nodes should now have an active channel with
+	// the created HTLC pending on all of them.
+	ht.AssertActiveHtlcs(alice, payHash[:])
+	ht.AssertActiveHtlcs(bob, payHash[:])
+	ht.AssertActiveHtlcs(carol, payHash[:])
+
+	// Wait for carol to mark invoice as accepted. There is a small gap to
+	// bridge between adding the htlc to the channel and executing the exit
+	// hop logic.
+	ht.AssertInvoiceState(stream, lnrpc.Invoice_ACCEPTED)
+
+	// Increase the fee estimate so that the following force close tx will
+	// be cpfp'ed.
+	ht.SetFeeEstimate(30000)
+
+	// At this point, Bob decides that he wants to exit the channel
+	// immediately, so he force closes his commitment transaction.
+	hasAnchors := commitTypeHasAnchors(c)
+	closeStream, _ := ht.CloseChannelAssertPending(
+		bob, aliceChanPoint, true,
+	)
+	bobForceClose := ht.AssertStreamChannelForceClosed(
+		bob, aliceChanPoint, hasAnchors, closeStream,
+	)
+
+	var expectedTxes int
+	switch c {
+	// Alice will sweep her commitment output immediately.
+	case lnrpc.CommitmentType_LEGACY:
+		expectedTxes = 1
+
+	// Alice will sweep her commitment and anchor output immediately.
+	case lnrpc.CommitmentType_ANCHORS:
+		expectedTxes = 2
+
+	// Alice will sweep her anchor output immediately. Her commitment
+	// output cannot be swept yet as it has incurred an additional CLTV due
+	// to being the initiator of a script-enforced leased channel.
+	case lnrpc.CommitmentType_SCRIPT_ENFORCED_LEASE:
+		expectedTxes = 1
+
+	default:
+		ht.Fatalf("unhandled commitment type %v", c)
+	}
+
+	ht.Miner.AssertNumTxsInMempool(expectedTxes)
+
+	// Suspend Bob to force Carol to go to chain.
+	restartBob := ht.SuspendNode(bob)
+
+	// Settle invoice. This will just mark the invoice as settled, as there
+	// is no link anymore to remove the htlc from the commitment tx. For
+	// this test, it is important to actually settle and not leave the
+	// invoice in the accepted state, because without a known preimage, the
+	// channel arbitrator won't go to chain.
+	carol.RPC.SettleInvoice(preimage[:])
+
+	// We'll now mine enough blocks so Carol decides that she needs to go
+	// on-chain to claim the HTLC as Bob has been inactive.
+	numBlocks := padCLTV(uint32(invoiceReq.CltvExpiry-
+		lncfg.DefaultIncomingBroadcastDelta) - 1)
```

Comment:
hawk eyes! Yeah they deserve to be in their own commits since they are actually fixing the current tests. Mostly it's an extra unwanted block that was mined, and they are now fixed. Meanwhile, when breaking the tests, I further emphasize this issue by using a variable `blocksMined` to track how many blocks are mined so far.